### PR TITLE
Feature/custom limits

### LINKED ISSUE #755: Custom limits values for dimensions
> Sometimes we want custom available limits. Per split or per datacube. Does it depend on visualisation?
From #472

- Dimension should have `limitValues` option. 
- If not provided, current hardcoded values should be used as default
- Resolvers still can set custom limits:
  - line/bar chart should limit to hard `COLORS_COUNT`
  - table, heat map should pick some available value

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**
Closes #755

### CODE REVIEW

File: src/common/models/dimension/dimension.ts [hunk 1]
```diff
@@ -214,12 +222,19 @@ export class Dimension implements Instance<DimensionValue, DimensionJS> {
...
       this.multiValue === other.multiValue &&
       this.url === other.url &&
+      this.limitsEqual(other.limits) &&
       this.granularitiesEqual(other.granularities) &&
       granularityEquals(this.bucketedBy, other.bucketedBy) &&
       this.bucketingStrategy === other.bucketingStrategy &&
       this.sortStrategy === other.sortStrategy;
   }
 
+  private limitsEqual(otherLimits: number[]): boolean {
```
Comment (mkuthan) [inline src/common/models/dimension/dimension.ts:232]: move to common lib (this is equal for number[] not limits)

File: src/common/view-definitions/version-4/split-definition.ts [hunk 2]
```diff
@@ -95,21 +96,21 @@ function fromSort(sort: Sort): SplitSortDefinition {
   return { ref, ...rest };
 }
 
-function toLimit(limit: unknown): number | null {
+function toLimit(limit: unknown, dimension: Dimension): number | null {
   if (limit === null) return null;
   if (isNumber(limit) && isFiniteNumber(limit)) return limit;
-  return AVAILABLE_LIMITS[0];
+  return dimension.limits[0];
```
Comment (mkuthan) [inline src/common/view-definitions/version-4/split-definition.ts:102]: encapsulate into dim.defaultLimit

**Review by mkuthan — CHANGES_REQUESTED** [review]
Very first comments into the draft PR :)

File: src/common/view-definitions/version-4/split-definition.ts (hunk 2 above)
Comment (adrianmroz-allegro) [reply src/common/view-definitions/version-4/split-definition.ts:102]: I can extract this with better name, but this is very specific behaviour for split definition converter. 
It is not "dimensions default limit value" but more like "I got malformed data and need to pick something".

I will keep an eye on this when implementing #756, where visualisation picks "default limit value" for split.

**Review by fpoon — APPROVED** [review]

**Pull request merged**